### PR TITLE
Release Dec 27, 2024 : Stable Diffusion 3, Exaone, etc

### PR DESCRIPTION
**Pull request by rebel-jongho**
## **Summary**

This pull request adds and updates multiple examples in the rbln-model-zoo, focusing on enhancing compatibility, fixing imports, improving formatting, and extending functionality across various model scripts. Key updates are summarized below:

## **Key Changes**

1. **Additions of New Models**:
   - **Stable Diffusion 3**: Added comprehensive examples for Text-to-Image (T2I), Inpainting, and Image-to-Image (I2I) models using `optimum-rbln`.
   - **YOLOv5-Face**: Introduced face detection models with `compile.py`, `inference.py`, and required dependencies.
   - **YOLOv8 Pose Estimation**: Integrated support for pose estimation models with wrapper classes for pose keypoints.
   - **EXAONE**: Added new configurations for EXAONE 3.5 (2.4B and 7.8B LLM models).

1. **vLLM Integration and Serving**:
    - Added scripts for vLLM-based inference to improve large language model (LLM) serving performance and scalability.
    - Created new or updated existing serving.py (or similarly named scripts) to show end-to-end serving pipelines. Included Dockerfile instructions (if applicable) and environment configurations to streamline deployment for inference serving.

1. **Formatting Improvements**:
   - Reformatted argument parsing for better readability in multiple scripts.
   - Updated code style to align with PEP8 standards across Python scripts.

1. **Dependency and Import Adjustments**:
   - Fixed import orders for clarity and compliance with common Python practices.
   - Removed unnecessary imports and organized required ones for efficiency.

1. **Etc**:
   - Updated model references to ensure compatibility with the latest versions of RBLN runtime and datasets.
   - Simplified default settings for some models like Llama and Mistral to remove redundant arguments.

### CODE REVIEW

File: huggingface/text2text-generation/exaone/exaone-3-7.8B/compile.py [hunk 1]
```diff
@@ -4,14 +4,6 @@
...
 
 def main():
-    """
-    EXAONE Model Usage License:
-    
-    - Solely for research purposes. This includes evaluation, testing, academic research, experimentation, 
-      and participation in competitions, provided that such participation is in a non-commercial context.
-    - For commercial use and larger context length, please contact LG AI Research, contact_us@lgresearch.ai
-    - Please refer to License Policy for detailed terms and conditions: https://huggingface.co/LGAI-EXAONE/EXAONE-3.0-7.8B-Instruct/blob/main/LICENSE
-    """
```
Comment (rebel-sunwook) [inline huggingface/text2text-generation/exaone/exaone-3-7.8B/compile.py:14]: 이부분 빠지면 안될꺼 같은데 확인부탁드립니다.

**Review by rebel-junamsong — APPROVED** [review]
lgtm

File: pytorch/vision/detection/yolov3/compile.py [hunk 2]
```diff
@@ -22,6 +23,12 @@ def parsing_argument():
 def main():
     args = parsing_argument()
     model_name = args.model_name
+
+    weight = model_name + ".pt"
+    torch.hub.download_url_to_file(
+        f"https://github.com/ultralytics/yolov5/releases/download/v7.0/{weight}", weight
+    )
+
```
Comment (rebel-shshin) [inline pytorch/vision/detection/yolov3/compile.py:31]: @rebel-sunwook 님 이 부분은 SSL error 인한 우회용 코드가 추가된건가요??

File: pytorch/vision/face_detection/yolov5-face/compile.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+import torch
+
+sys.path.append(os.path.join(sys.path[0], "yolov5-face"))
+from models.experimental import attempt_load
+
+pt_file_urls = {
+    "yolov5n-0.5": "https://drive.google.com/uc?id=1XJ8w55Y9Po7Y5WP4X1Kg1a77ok2tL_KY",
+    "yolov5n-face": "https://drive.google.com/uc?id=18oenL6tjFkdR1f5IgpYeQfDFqU4w3jEr",
+    "yolov5s-face": "https://drive.google.com/uc?id=1zxaHeLDyID9YU4-hqK7KNepXIwbTkRIO",
+    "yolov5m-face": "https://drive.google.com/uc?id=1Sx-KEGXSxvPMS35JhzQKeRBiqC98VDDI",
```
Comment (rebel-shshin) [inline pytorch/vision/face_detection/yolov5-face/compile.py:16]: 여기 적혀있는 주소는 우리가 올린건 아니고 original yolov5-face repo에서 올려둔 웨이트 맞을까요??

Comment (rebel-shshin) [reply pytorch/vision/face_detection/yolov5-face/compile.py:16]: @rebel-kblee

Comment (rebel-shshin) [reply pytorch/vision/face_detection/yolov5-face/compile.py:16]: @rebel-sunwook

File: pytorch/vision/pose_estimation/motionbert/inference.py [hunk 4]
```diff
@@ -29,7 +28,7 @@ def parsing_argument():
         type=str,
         default="base",
         choices=["base", "lite"],
-        help="(str) type, motionBERT 3D pose model name.",
+        help="(str) type, motionbert 3dpose model_name.",
```
Comment (rebel-shshin) [inline pytorch/vision/pose_estimation/motionbert/inference.py:31]: 수정 전이 더 맞을 것 같습니다.

File: pytorch/vision/detection/yolov3/compile.py (hunk 2 above)
Comment (rebel-sunwook) [reply pytorch/vision/detection/yolov3/compile.py:31]: 이건 SSL error는 아니고, yolov3의 attempt_load에서 알수 없는 이유로 yolov3.pt 파일을 다운로드 하지 않는 이슈가 있어서, attempt_load 전에 direct로 install하는 단계입니다. 들어가도 무방한 코드 입니다. cc. @rebel-junamsong

File: pytorch/vision/face_detection/yolov5-face/compile.py (hunk 3 above)
Comment (rebel-sunwook) [reply pytorch/vision/face_detection/yolov5-face/compile.py:16]: 넵 original yolov5-face에서 올려둔 weight 입니다. 이거도 LISENCE 같은거 추가해야 될까요?? https://github.com/deepcam-cn/yolov5-face 여기 README.md에 기술된 링크를 참조한 것 뿐이긴 합니다.

File: huggingface/stable-diffusion/stable_diffusion_3_i2i/inference.py [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    pipe = RBLNStableDiffusion3Img2ImgPipeline.from_pretrained(
+        model_id=os.path.basename(model_id),
+        export=False,
+        rbln_config={
+            "text_encoder": {"device": 1},
+            "text_encoder_2": {"device": 1},
+            "text_encoder_3": {"device": 0},
+            "transformer": {"device": 1},
+            "vae": {"device": 1},
+        },
```
[the remark this answers is not in the document]
Comment (rebel-jongho) [reply huggingface/stable-diffusion/stable_diffusion_3_i2i/inference.py:35]: @rebel-kblee fyi

File: pytorch/vision/face_detection/yolov5-face/compile.py (hunk 3 above)
Comment (rebel-shshin) [reply pytorch/vision/face_detection/yolov5-face/compile.py:16]: 따로 license는 필요없을 것 같습니다 :)

File: pytorch/vision/pose_estimation/motionbert/compile.py [hunk 6]
```diff
@@ -28,7 +28,7 @@ def parsing_argument():
         type=str,
         default="base",
         choices=["base", "lite"],
-        help="(str) type, motionBERT 3D pose model name.",
+        help="(str) type, motionbert 3dpose model_name.",
```
Comment (rebel-shshin) [inline pytorch/vision/pose_estimation/motionbert/compile.py:31]: @rebel-jongho 님 internal/external sync 맞춰야하는부분이 하나 더 있는 것 같네요. 아까는 inference.py만 수정된것 같습니다.

**Review by rebel-sunwook — APPROVED** [review]
👍

**Pull request merged**